cmd/util/ledger/migrations/ordered_map_migration_test.go

```diff
@@ -0,0 +1,278 @@
+
+		migrated := &OrderedMapMigration{}
+		migrated.initPersistentSlabStorage(NewView(migratedPayload))
+		migrated.initIntepreter()
+
+		stored := migrated.Interpreter.ReadStored(cadenceAddress, "public", "Bar")
+		require.Equal(t, stored, one)
+
+		stored = migrated.Interpreter.ReadStored(cadenceAddress, "storage", "Foo")
+		require.IsType(t, &interpreter.ArrayValue{}, stored)
+
+		stored = migrated.Interpreter.ReadStored(cadenceAddress, "storage", "Bar")
+		require.Equal(t, stored, str)
+
+		stored = migrated.Interpreter.ReadStored(cadenceAddress, "private", "Bar")
+		require.IsType(t, &interpreter.DictionaryValue{}, stored)
+
+		stored = migrated.Interpreter.ReadStored(cadenceAddress, "storage", "Baz")
+		require.Equal(t, stored, nil)
+	})
+}
+
+func TestMultipleAccounts(t *testing.T) {
+	dir := t.TempDir()
+	mig := OrderedMapMigration{
+		Log:       zerolog.Logger{},
+		OutputDir: dir,
+	}
+
+	address1 := flow.HexToAddress("0x1")
+	cadenceAddress1, _ := common.HexToAddress("0x1")
+
+	address2 := flow.HexToAddress("0x2")
+	cadenceAddress2, _ := common.HexToAddress("0x2")
+
+	address3 := flow.HexToAddress("0x3")
+	cadenceAddress3, _ := common.HexToAddress("0x3")
+
+	mig.initialize([]ledger.Payload{
+		{Key: createAccountPayloadKey(address1, state2.KeyExists), Value: []byte{1}},
+		{Key: createAccountPayloadKey(address1, state2.KeyStorageUsed), Value: utils.Uint64ToBinary(1)},
+		{Key: createAccountPayloadKey(address2, state2.KeyExists), Value: []byte{1}},
+		{Key: createAccountPayloadKey(address2, state2.KeyStorageUsed), Value: utils.Uint64ToBinary(1)},
+		{Key: createAccountPayloadKey(address3, state2.KeyExists), Value: []byte{1}},
+		{Key: createAccountPayloadKey(address3, state2.KeyStorageUsed), Value: utils.Uint64ToBinary(1)},
+	})
+
+	encodeValue := func(v interpreter.Value, address flow.Address) ledger.Value {
+		storable, _ := v.Storable(mig.newStorage, atree.Address(address), math.MaxUint64)
+		encodedInt, _ := atree.Encode(storable, interpreter.CBOREncMode)
+		return encodedInt
+	}
+
+	t.Run("sort values", func(t *testing.T) {
+
+		one := interpreter.NewIntValueFromInt64(1)
+		two := interpreter.NewIntValueFromInt64(1)
+		three := interpreter.NewIntValueFromInt64(1)
+		four := interpreter.NewIntValueFromInt64(1)
+		five := interpreter.NewIntValueFromInt64(1)
+		six := interpreter.NewIntValueFromInt64(1)
+
+		payload := []ledger.Payload{
+			{Key: createAccountPayloadKey(address1, "storage\x1fFoo"), Value: encodeValue(one, address1)},
+			{Key: createAccountPayloadKey(address2, "storage\x1fFoo"), Value: encodeValue(two, address2)},
+			{Key: createAccountPayloadKey(address3, "storage\x1fFoo"), Value: encodeValue(three, address3)},
+			{Key: createAccountPayloadKey(address1, "storage\x1fBar"), Value: encodeValue(four, address1)},
+			{Key: createAccountPayloadKey(address1, "storage\x1fBaz"), Value: encodeValue(five, address1)},
+			{Key: createAccountPayloadKey(address3, "public\x1fFoo"), Value: encodeValue(six, address3)},
+		}
```

Comment:
This example data here looks like it correctly represents the current encoding scheme: Paths are encoded like this, and the values are encoded storables.

However, the test cases here only cover the fixed-size `Int64` type. As the encoding of the values to storables differs depending on type and size, could you please also add test cases / example payloads where the storable representation is not the inlined encoded data?

For example, a small integer or string is directly stored "inline" in the value. However, large immutable values (like large strings) and especially mutable Cadence values / containers, like Cadence arrays, dictionaries, and composites, are actually stored in a separate register/slab, and the "top-level"/"entry" registers here are just `StorageIDStorable`s, effectively "pointing" to the registers holding the actual data.

Reply:
I thought the previous test checked this case, but it's possible I am wrong. Does creating and storing an array/dictionary not exercise this case?